internal/kafka/internal/workers/kafka_mgrs/promotion/internal/actions/reserve_desired_quota_action.go

```diff
@@ -0,0 +1,39 @@
+package actions
+
+import (
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/internal/kafka/internal/api/dbapi"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/internal/kafka/internal/config"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/internal/kafka/internal/services"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/internal/kafka/internal/workers/kafka_mgrs/promotion/chain"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/api"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/errors"
+)
+
+var _ chain.ReconcileAction[PromotionContext] = &ReserveDesiredQuotaAction{}
+
+type ReserveDesiredQuotaAction struct {
+	quotaServiceFactory services.QuotaServiceFactory
+	kafkaConfig         config.KafkaConfig
+}
+
+func NewReserveDesiredQuotaAction(kafkaConfig config.KafkaConfig, quotaServiceFactory services.QuotaServiceFactory) chain.ReconcileAction[PromotionContext] {
+	return &ReserveDesiredQuotaAction{
+		quotaServiceFactory: quotaServiceFactory,
+		kafkaConfig:         kafkaConfig,
+	}
+}
+
+func (d *ReserveDesiredQuotaAction) PerformJob(kafkaRequest *dbapi.KafkaRequest, currentResult chain.Result[PromotionContext]) (chain.Result[PromotionContext], bool, error) {
```

Comment:
Would be good to document the behavior of the different implementations of quota actions. Specially if they are idempotent